File: python/Vision/ComputerVision2.1.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

import json
import requests
import os
from pprint import pprint

'''
This sample makes a call to the Face API with a URL image query
to detect faces and features in an image.
Face API: https://westus.dev.cognitive.microsoft.com/docs/services/563879b61984550e40cbbe8d/operations/563879b61984550f30395236
'''

# Add your Face subscription key and endpoint to your environment variables.
subscription_key = os.environ['FACE_SUBSCRIPTION_KEY']
endpoint = os.environ['FACE_ENDPOINT'] + '/face/v1.0/detect'

# Request headers.
headers = {
    'Content-Type': 'application/json',
    'Ocp-Apim-Subscription-Key': subscription_key,
}

# Request parameters.
params = {
    'returnFaceId': 'true',
    'returnFaceLandmarks': 'false',
    'returnFaceAttributes': 'age,gender,headPose,smile,facialHair,glasses,emotion,hair,makeup,occlusion,accessories,blur,exposure,noise',
}

# The URL of a JPEG image of faces to analyze.
body = {'url': 'https://raw.githubusercontent.com/Azure-Samples/cognitive-services-sample-data-files/master/Face/images/test-image-person-group.jpg'}

try:
    # Call API. 
    response = requests.post(endpoint, headers=headers, params=params, json=body)
    response.raise_for_status()

    print("\nHeaders:\n")
    print(response.headers)

    print("\nJSON Response:\n")
    pprint(response.json())

except Exception as e:
    print('Error:')
    print(e)

File: java/Vision/ComputerVision2.1.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import java.net.URI;
import org.apache.http.HttpEntity;
import org.apache.http.HttpResponse;
import org.apache.http.client.HttpClient;
import org.apache.http.client.methods.HttpPost;
import org.apache.http.entity.StringEntity;
import org.apache.http.client.utils.URIBuilder;
import org.apache.http.impl.client.DefaultHttpClient;
import org.apache.http.util.EntityUtils;
import org.json.JSONObject;

/**
 * Include these library jars (or newer +), add to a lib folder:
 *   commons-logging-4.0.6+ 
 *   httpclient-4.5.3+ 
 *   httpcore-4.4.13+ 
 *   json-20190722+
 *
 * To compile and run from command line: 
 *   javac Main.java -cp .;lib\*
 *   java -cp .;lib\* Main
 */

public class Main
{
  // Add your Computer Vision subscription key and endpoint to your environment variables.
    private static String subscriptionKey = System.getenv("COMPUTER_VISION_SUBSCRIPTION_KEY");
    private static String baseUrl = System.getenv("COMPUTER_VISION_ENDPOINT");

    private static final String endpoint = baseUrl + "/vision/v2.1/analyze";

    // Image of an entity (well-known place)
    private static String image = "https://raw.githubusercontent.com/Azure-Samples/cognitive-services-sample-data-files/master/ComputerVision/Images/landmark.jpg";

    public static void main(String[] args)
    {
        HttpClient httpclient = new DefaultHttpClient();

        try
        {
            URIBuilder builder = new URIBuilder(endpoint);

            // Request parameters. All of them are optional.
            builder.setParameter("visualFeatures", "Categories,Description,Color");
            builder.setParameter("language", "en");

            // Prepare the URI for the REST API call.
            URI uri = builder.build();
            HttpPost request = new HttpPost(uri);

            // Request headers.
            request.setHeader("Content-Type", "application/json");
            request.setHeader("Ocp-Apim-Subscription-Key", subscriptionKey);

            // Request body.
            StringEntity reqEntity = new StringEntity("{\"url\": \"" + image + "\"}");
            request.setEntity(reqEntity);

            // Execute the REST API call and get the response entity.
            HttpResponse response = httpclient.execute(request);
            HttpEntity entity = response.getEntity();

            if (entity != null)
            {
                // Format and display the JSON response.
                String jsonString = EntityUtils.toString(entity);
                JSONObject json = new JSONObject(jsonString);
                System.out.println("REST Response:\n");
                System.out.println(json.toString(2));
            }
        }
        catch (Exception e)
        {
            // Display error message.
            System.out.println(e.getMessage());
        }
    }  

}
